src/PrestaShopBundle/Controller/Admin/Improve/International/LocalizationController.php

```diff
@@ -0,0 +1,193 @@
+     *
+     * @return RedirectResponse
+     */
+    public function saveSettingsAction(Request $request)
+    {
+        $legacyController = $request->attributes->get('_legacy_controller');
+
+        if ($this->isDemoModeEnabled()) {
+            return $this->redirectToRoute('admin_localization_show_settings');
+        }
+
+        if (!in_array(
+            $this->authorizationLevel($legacyController),
+            [
+                PageVoter::LEVEL_UPDATE,
+                PageVoter::LEVEL_CREATE,
+                PageVoter::LEVEL_DELETE,
+            ]
+        )) {
+            $this->addFlash('error', $this->trans('You do not have permission to edit this', 'Admin.Notifications.Error'));
+
+            return $this->redirectToRoute('admin_localization_show_settings');
+        }
+
+        $localizationFormHandler = $this->getLocalizationFormHandler();
+
+        $localizationForm = $localizationFormHandler->getForm();
+        $localizationForm->handleRequest($request);
+
+        if ($localizationForm->isSubmitted()) {
+            $data = $localizationForm->getData();
+
+            $errors = $localizationFormHandler->save($data);
+            if (empty($errors)) {
+                $this->addFlash('success', $this->trans('Update successful', 'Admin.Notifications.Success'));
+
+                return $this->redirectToRoute('admin_localization_show_settings');
+            }
+
+            $this->flashErrors($errors);
+        }
+
+        return $this->redirectToRoute('admin_localization_show_settings');
+    }
+
+    /**
+     * Handles localization pack import
+     *
+     * @param Request $request
+     *
+     * @return RedirectResponse
+     */
+    public function importLocalizationPackAction(Request $request)
+    {
+        if ($this->isDemoModeEnabled()) {
+            return $this->redirectToRoute('admin_localization_show_settings');
+        }
+
+        $localizationPackImportForm = $this->createForm(ImportLocalizationPackType::class);
+        $localizationPackImportForm->handleRequest($request);
+
+        if ($localizationPackImportForm->isSubmitted()) {
+            $data = $localizationPackImportForm->getData();
+
+            $localizationImportConfig = new LocalizationPackImportConfig(
+                $data['iso_localization_pack'],
+                $data['content_to_import'],
+                $data['download_pack_data']
+            );
+
+            $localizationPackImporter = $this->get('prestashop.core.localization.pack.import.importer');
+            $errors = $localizationPackImporter->import($localizationImportConfig);
+
+            if (empty($errors)) {
+                $this->addFlash(
+                    'success',
+                    $this->trans('Localization pack imported successfully.', 'Admin.International.Notification')
+                );
+
+                return $this->redirectToRoute('admin_localization_show_settings');
+            }
+
+            foreach ($errors as $error) {
```

Comment:
you can use `flashErrors(array $errors)` from `FrameworkBundleAdminController`

Reply:
the problem with `flashErrors(array $errors)` is that it adds errors like:
`$this->addFlash('error', $this->trans($error['key'], $error['domain'], $error['parameters']));`
 in my case, i already have translated errors.

What do you think if we updated `flashErrors(array $errors)` to this:

```php
foreach ($errorMessages as $error) {
    if (is_string($error)) {
        $message = $error;
    else {
        $message = $this->trans($error['key'], $error['domain'], $error['parameters']);
    }
    $this->addFlash('error', $message);
}
```

Reply:
why your errors messages are already translated? 

hmm, this is not supposed to happens ^^ I'll think about it, answer tomorrow ;)